### ISSUE TITLE
Feature Request - FDroid

### ISSUE THREAD

**Issue by Stele77** [Chooloo/call_manage#3]
**Is your feature request related to a problem? Please describe.**
Problem is, that a open source App would be be greatly suited for the FOSS Fdroid store, so privacy consious and open source supporters dont need to trust and use the Google Play Store.

**Describe the solution you'd like**
Publishing this App on FDroid

**Describe alternatives you've considered**
APK downloads can in the meantime help out those who dont want to use play store and give their data and privacy to Google. But only in FDroid can the app be auto updated, instead of manually downloading and installing APK files.

**Additional context**
FDroid is the standard way for open source Android App. The open source community expects this on Android; Plus it would greatly help this App to become more well known in the community and to new users.
As there are no good open source dialers out there, this App would be a huge benefit for anyone who wants to use Android as open sourcy and privacy minded as possible.

**Comment by Poussinou** [comment]
If you need it, you can open a Request For Packaging [here](https://gitlab.com/fdroid/rfp/issues), or I can do it for you. You will receive some help from our side at F-Droid.

**Comment by BasilTomato** [comment]
I have opened a RFP issue at F-Droid: https://gitlab.com/fdroid/rfp/-/issues/1285

First request is: could you establish Fastlane file structure in the app's repo? Details regarding the structure: https://gitlab.com/snippets/1895688

**Comment by roeiedri** [comment]
Unfortunately we dont really have the time to handle this
I saw the link you sent and at a glance i couldn't really understand the purpose of it
Tough i want to help, can u maybe elaborate about what Fastlane file is and maybe help so i can make it happen faster?
:)

**Comment by IzzySoft** [comment]
Aw, @roeiedri – adding 2 files shouldn't take that much time. As the snippet says: the mandatory 2 files are `short_description.txt` with a summary not longer than 80 chars, and a `full_description.txt` with a more detailed description, both in `fastlane/metadata/android/en-US`. Everything else is optional, though screenshots are highly recommended.

Ah well, there you go: #118 :wink:

**Comment by offa** [comment]
Any news on this?

**Comment by roeiedri** [comment]
App still in development :)

**Comment by offa** [comment]
Yeah, but how's F-Droid going? :smile:

**Comment by roeiedri** [comment]
oh haha I didn't event started that, does it take time to approve or something?

**Comment by offa** [comment]
The first step is pending already: #118 :smile:

**Comment by realmain** [comment]
If we can get this app into fastlane, it'll be one step closer to getting into F-Droid, which would give this app more presence.

**Comment by fossdd** [comment]
Yep, would nice if this could be merged. The app itself looks good to be packaged. After that it would needed to create a new [GitHub release](https://github.com/Chooloo/koler/releases)

**Comment by IzzySoft** [comment]
Fastlane got merged \o/ You wanna pick up at RFP, @fossdd?

**Comment by FestplattenSchnitzel** [comment]
Hello,

currently we cannot get https://gitlab.com/fdroid/rfp/-/issues/1285 done, as the following isn't done here yet:

- [x] Add Fastlane structures to the repository as described [here](https://gitlab.com/-/snippets/1895688).
- [ ] Cover the above change(s) under a new or existing tag.

If there is no activity here or in the F-Droid issue in the next 4 weeks, we will close our issue to reduce the number of open issues.
In our [wiki](https://gitlab.com/fdroid/wiki/-/wikis/Internal/Issue-Cleanup-Policy) you can see more details about this process.

Please contact us when you need help.

Thank you for helping us spreading FOSS apps in the world :)

**Comment by proletarius101** [comment]
Seems good enough to be included: https://gitlab.com/fdroid/fdroiddata/-/merge_requests/10486

**Comment by offa** [comment]
On F-Droid finally: https://f-droid.org/app/com.chooloo.www.koler

**Comment by IzzySoft** [comment]
@roeiedri may I ask if you want me to keep Koler in my repo then? Usually I remove apps when they've been accepted to F-Droid, after giving them a little overlap (about 10 days) so people notice.

**Comment by roeiedri** [comment]
Yeah i want to upload it on my own i just don't find the time, hopefully this weekend @IzzySoft

**Comment by IzzySoft** [comment]
@roeiedri so that means you want to keep it in my repo? No problem, I then just need to mark it accordingly. So please state a clear Nes or Yo – err, Yes or No :smile:

**Comment by roeiedri** [comment]
Wait lol i meant i would like to have it under a Chooloo repo

**Comment by realmain** [comment]
Can probably close this issue now that Koler is in the official F-droid repo. Looks like everything is set up correctly.

**Comment by IzzySoft** [comment]
> i would like to have it under a Chooloo repo

Ah, so you want to [setup your own repo](https://android.izzysoft.de/articles/named/fdroid-simple-binary-repo)? Well, go ahead then :smiley: I'll then remove Chooloo/Koler from my repo after about 10 days of overlap with F-Droid, as usual.

PS: That article will probably get a sequel with some fine-tunings (how to integrate screenshots, and things like that) as soon as time permits. Unlikely within the next couple of weeks, but in my plans.

**Comment by roeiedri** [comment]
Yes @IzzySoft 
Thank you!!

**Comment by IzzySoft** [comment]
Marked, thanks!

**Comment by roeiedri** [comment]
@IzzySoft May I ask for some help with F-Droid? I don't understand the path to uploading Koler.
Do I need to setup an F-Droid server in my PC?

**Comment by roeiedri** [comment]
@IzzySoft I want to setup a repo in order to upload Koler.
Also yes, idk how updates get to F-Droid :)

**Comment by IzzySoft** [comment]
Updates for F-Droid are usually picked up whenever you tag a new release (provided you stick to your tag name patterns, and `versionCode` was increased). A process called CheckUpdates takes care for that. For your app, according to the MR this is defined as

```yaml
AutoUpdateMode: Version v%v
UpdateCheckMode: Tags
UpdateCheckData: koler/build.gradle|versionCode\s(\d+)||
```

So tag names should be `v<versionName>` for this to work.

As for your own repo, what part is missing from the linked article? If everything has been set up (including the metadata), updates are as easy as dropping the APK into the `repo/` directory, and then run `fdroid update` and `fdroid deploy`. It's mentioned [in this section](https://android.izzysoft.de/articles/named/fdroid-simple-binary-repo#feeding):

> Copy some `.apk` files into your new repo (make sure to place them inside repo/ […] On subsequent runs when only updating descriptions or apps, you’ll only need to run `fdroid update`.

The `deploy` part is in the next section. Btw, the docker container mentioned in references even automates that: you then only need to drop the APK in, and the container takes care for the rest (runs the `update` once per hour if I understood Sylvia correctly).

**Comment by yephny** [comment]
Even if you set up your own repository, please do not remove it from the official F-Droid repo, as they build all apps themselves, which makes them the most trusted source for apks. Currently, it is removed from Izzy's per your request, leaving no way for easy updates.

**Comment by freshgum-bubbles** [comment]
Ideally, Koler would remain in the main F-Droid repository. If you're looking for faster updates (though I don't understand *why*) NewPipe actually run their *own* repository **alongside** the F-Droid one. The custom one gets faster builds. Custom repositories are not monitored by F-Droid, of course, and therefore carry none of the security benefits the main repository offers, such as building the apps themselves, ensuring anti-features are correct, etc. Bear in mind that this is all extra burden on *you* if you were to run your own repository.

Is there any reason as to why you'd like to roll your own repository, or are you just experimenting with F-Droid tech ...?

**Comment by IzzySoft** [comment]
@freshgum-bubbles don't forget that apps like NewPipe (and also CCTG plus some more) use "reproducible builds". This means, taken their APK and the ones built by F-Droid and stripping both of their certificates/signing elements, they are "binary identical". As long as that's confirmed (i.e. whenever they match), F-Droid takes the signed APK from the resp. devs and places its own signing on top – so they are "double-signed".

What does that mean? I install from one side, and update wherever the APK shows up first. Or I always go with F-Droid unless there's some "urgent update" I want ASAP, which I can then take from the dev directly. In other words: cross-updating is possible. This way you have the best of both worlds: your most trustworthy source (F-Droid), and immediate security updates should F-Droid take too long.

Another point for a separate repo: "early birds and testers". There are several apps in my repo which are also on F-Droid – where F-Droid gets only the stable versions, while mine picks (only the) betas/pre-releases. Benefit: Early adopters have an easy way to stay up-to-date, bugs are found fast and get fixed – while all other users can rely on safe and stable releases via F-Droid.

**TL;DR:** Yes, Koler should stay on F-Droid (that was the goal of this issue). But a "secondary repo" makes a lot of sense, too – for the outlined approaches.

**Comment by shuvashish76** [comment]
I agree with @freshgum-bubbles that ,  NewPipe's own repository makes sense cause their users need instant fix _(usually F-Droid takes 4-7 days for building the apk from source)_ in case YouTube change something which break NewPipe.

Check their recent blog here: https://newpipe.net/blog/pinned/release/newpipe-0.21.16-hotfix/

I don't think custom repo is necessary just for faster Koler app updates.

**Comment by freshgum-bubbles** [comment]
> don't forget that apps like NewPipe (and also CCTG plus some more) use "reproducible builds". This means, taken their APK and the ones built by F-Droid and stripping both of their certificates/signing elements, they are "binary identical". As long as that's confirmed (i.e. whenever they match), F-Droid takes the signed APK from the resp. devs and places its own signing on top – so they are "double-signed".

Yeah, I wasn't 100% on whether F-Droid was doing that, so I didn't mention it. Thank you for confirming, Izzy :)

> I don't think custom repo is necessary just for faster Koler app updates.

That feels like a tempered conclusion. In theory, a phone app shouldn't need the immediacy of updates from a third-party repository. Perhaps something like Izzy mentioned above, re double-signing, would work for Koler?

**Comment by IzzySoft** [comment]
Note that reproducible builds need some extra efforts. Every now and then one of those fails to reproduce (happened to both apps I've mentioned, and at least one more I remember). It's definitely a good goal to target though, don't get me wrong.
